aws/resource_aws_athena_database.go

```diff
@@ -38,18 +38,73 @@ func resourceAwsAthenaDatabase() *schema.Resource {
 				Optional: true,
 				Default:  false,
 			},
+			"encryption_key": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
```

Comment:
Two things:
* The API refers to this as `encryption_option` -- is there any reason to diverge from their naming?
* We should validate this attribute using the SDK provided constants:

```go
ValidateFunc: validation.StringInSlice([]string{
  athena.EncryptionOptionCseKms,
  athena.EncryptionOptionSseKms,
  athena.EncryptionOptionSseS3,
}, false),
```

Reply:
Good call on the validate. I mentioned why I diverged from the API above. I think the HCL I've proposed would look cleaner than this:

```hcl
encryption_configuration {
  encryption_option = "CSE_KMS"
  kms_key = "${aws_kms_key.foo.id}"
}
```

It feels verbose to me. I think the CodePipeline API looks/feels more correct.